{
  "schema_version": "1.5.0",
  "id": "BIT-tensorflow-2021-37675",
  "details": "TensorFlow is an end-to-end open source platform for machine learning. In affected versions most implementations of convolution operators in TensorFlow are affected by a division by 0 vulnerability where an attacker can trigger a denial of service via a crash. The shape inference [implementation](https://github.com/tensorflow/tensorflow/blob/460e000de3a83278fb00b61a16d161b1964f15f4/tensorflow/core/framework/common_shape_fns.cc#L577) is missing several validations before doing divisions and modulo operations. We have patched the issue in GitHub commit 8a793b5d7f59e37ac7f3cd0954a750a2fe76bad4. The fix will be included in TensorFlow 2.6.0. We will also cherrypick this commit on TensorFlow 2.5.1, TensorFlow 2.4.3, and TensorFlow 2.3.4, as these are also affected and still in supported range.",
  "aliases": [
    "CVE-2021-37675"
  ],
  "affected": [
    {
      "package": {
        "ecosystem": "Bitnami",
        "name": "tensorflow",
        "purl": "pkg:bitnami/tensorflow"
      },
      "severity": [
        {
          "type": "CVSS_V3",
          "score": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H"
        }
      ],
      "ranges": [
        {
          "type": "SEMVER",
          "events": [
            {
              "introduced": "2.3.0"
            },
            {
              "fixed": "2.3.4"
            },
            {
              "introduced": "2.4.0"
            },
            {
              "fixed": "2.4.3"
            }
          ]
        },
        {
          "type": "SEMVER",
          "events": [
            {
              "introduced": "2.5.0"
            },
            {
              "last_affected": "2.5.0"
            },
            {
              "introduced": "2.6.0-rc0"
            },
            {
              "last_affected": "2.6.0-rc0"
            },
            {
              "introduced": "2.6.0-rc1"
            },
            {
              "last_affected": "2.6.0-rc1"
            },
            {
              "introduced": "2.6.0-rc2"
            },
            {
              "last_affected": "2.6.0-rc2"
            }
          ]
        }
      ]
    }
  ],
  "database_specific": {
    "severity": "Medium",
    "cpes": [
      "cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*",
      "cpe:2.3:a:google:tensorflow:2.5.0:*:*:*:*:*:*:*",
      "cpe:2.3:a:google:tensorflow:2.6.0:rc0:*:*:*:*:*:*",
      "cpe:2.3:a:google:tensorflow:2.6.0:rc1:*:*:*:*:*:*",
      "cpe:2.3:a:google:tensorflow:2.6.0:rc2:*:*:*:*:*:*"
    ]
  },
  "references": [
    {
      "type": "WEB",
      "url": "https://github.com/tensorflow/tensorflow/commit/8a793b5d7f59e37ac7f3cd0954a750a2fe76bad4"
    },
    {
      "type": "WEB",
      "url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-9c8h-2mv3-49ww"
    }
  ],
  "published": "2023-11-06T09:11:36.201Z",
  "modified": "2023-11-07T16:28:29.749Z"
}